Sample/ECommerce.Equinox/ECommerce.Api/Controllers/ShoppingCartsController.fs

```diff
@@ -0,0 +1,74 @@
+﻿namespace ECommerce.Api.Controllers
+
+open ECommerce.Domain
+open Microsoft.AspNetCore.Mvc
+open System
+
+type InitializeShoppingCartRequest = { clientId : Guid Nullable }
+type AddProductRequest = { productId : Guid; quantity : int }
+type RemoveProductRequest = { productId : Guid; price : decimal }
+
+[<ApiController>]
+[<Route("api/[controller]")>]
+type ShoppingCartsController(carts : ShoppingCart.Service) =
+    inherit ControllerBase()
+
+    [<HttpPost>]
+    member _.InitializeCart([<FromBody>] request : InitializeShoppingCartRequest) : Async<IActionResult> = async {
+        if obj.ReferenceEquals(null, request) then nameof request |> nullArg
+
+        // TODO this should be resolved via an idempotent lookup in the Client's list
```

Comment:
**QUESTION:** Why do we need to do lookup here?

Reply:
I don't mean literally here. I should have put it more completely in the comment, but IMO it's more realistic to model something closer to a real world app, i.e.
- client keeps (only) a client id as a cookie or similar
- initial shopping is anonymous, and that clientid can be used to look it up. When you want to add to cart, you can add with just the client id, or if you have the current valid cartid, fine, pass that.
- then, when you log in, the real identity is known, and normally uou merge the anon cart into a concrete cart -- which may or may not be initialized or closed, but you dont care
- when you log in, you the supply the clientid that belongs to the account back to the user
- you don't want to produce random orphan cartids as the very last thing you want soemone to do is take something out of a cart or forget they had it
- clients don't really care about the initializatiion (and forcing an extra roundtrip or flag when somone just wants to add to cart does not make sense either)

This implies two things:
- an indirection via clientid when accessing a cart
- the job of an add to cart function is to add it to the current cart, or start a new cart, but really the clientid is the main thing, and forcing initialization busywork roundtrips on clients does not help anyone
- the job of a confirm cart function is however relative to a particular cart - i.e. you do need some sort of protocol beyond the customer only ever passing a clientid

How flowery you want to make it is up to you of course, but the bottom line is that this call is not and can not be idempotent and is prone to creating lots of orphan carts that are zero use to anyone, and that's rarely a good design

Reply:
@oskardudycz I've extended the TODO comment and added a reference to this in the overview. Can you resolve this comment if and when you feel that covers it?

By the same token, if you have time to to and from on PMing the features, we could extend the logic to have a client maintain a cart history and have the commands insert into the current cart. Bonus points if there's going to be >1 sample in the repo doing the same thing...